src/libraries/System.Memory/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -576,64 +594,89 @@ private static unsafe void Ssse3Decode(ref byte* srcBytes, ref byte* destBytes,
                 8, 14, 13, 12,
                 -1, -1, -1, -1);
 
-            Vector128<sbyte> mask2F = Vector128.Create((sbyte)'/');
-            Vector128<sbyte> mergeConstant0 = Vector128.Create(0x01400140).AsSByte();
+            Vector128<byte> mask2F = Vector128.Create((byte)'/');
+            Vector128<byte> mergeConstant0 = Vector128.Create(0x01400140).AsByte();
             Vector128<short> mergeConstant1 = Vector128.Create(0x00011000).AsInt16();
-            Vector128<sbyte> zero = Vector128<sbyte>.Zero;
+            Vector128<byte> zero = Vector128<byte>.Zero;
+            Vector128<byte> one = Vector128.Create((byte)1);
+            Vector128<byte> f = Vector128.Create((byte)0xf);
 
             byte* src = srcBytes;
             byte* dest = destBytes;
 
             //while (remaining >= 24)
             do
             {
-                AssertRead<Vector128<sbyte>>(src, srcStart, sourceLength);
-                Vector128<sbyte> str = Sse2.LoadVector128(src).AsSByte();
+                Vector128<byte> str = Vector128.LoadUnsafe(ref *src);
 
                 // lookup
-                Vector128<sbyte> hiNibbles = Sse2.And(Sse2.ShiftRightLogical(str.AsInt32(), 4).AsSByte(), mask2F);
-                Vector128<sbyte> loNibbles = Sse2.And(str, mask2F);
-                Vector128<sbyte> hi = Ssse3.Shuffle(lutHi, hiNibbles);
-                Vector128<sbyte> lo = Ssse3.Shuffle(lutLo, loNibbles);
+                Vector128<byte> hiNibbles = Vector128.ShiftRightLogical(str.AsInt32(), 4).AsByte();
+                Vector128<byte> loNibbles = str;
+                if (Ssse3.IsSupported)
+                {
+                    hiNibbles = Vector128.BitwiseAnd(hiNibbles, mask2F);
+                    loNibbles = Vector128.BitwiseAnd(str, mask2F);
+                }
```

Comment:
Could you please explain why there are extra BitwiseAnd here for SSSE3 only?

Reply:
There's something more to it - It looks like the hiNibbles And is needed for this line:
                Vector128<byte> shift = SimdShuffle(lutShift.AsByte(), Vector128.Add(eq2F, hiNibbles), f);
Without it the testing breaks.